### ISSUE TITLE
Annoying info on abort scan

### ISSUE THREAD

**Issue by cmft** [sardana-org/sardana#1683]
BL11 users considered  annoying the info reported during a scan abort.
![image](https://user-images.githubusercontent.com/6168722/130965795-c2968bcd-33b9-47c8-8d52-36fb248b8137.png)
If this is the default behavior it could be move them to debug and report just the exception  (to be possible at the end of the chunks)
```
   def __stopObjects(self):
             .... 
               self.output(
                    "Stopping {} reserved by {}".format(obj, macro._name))
```

**Comment by reszelaz** [comment]
Hi @cmft,
I agree that what you observed was not correct and was annoying.
This issue should be already fixed with https://github.com/sardana-org/sardana/pull/1577 and https://github.com/sardana-org/sardana/pull/1579 (latest develop branch). The current situation is that all the reserved objects, which are automatically stopped/aborted by the MacroServer when the macro is stopped/aborted, are logged. In the case of an ascan this is one log for a motor and one for a measurement group. If you are not happy with this then please re-open.
Many thanks!

**Comment by cmft** [comment]
Hi @reszelaz ,
I tested with develop and I'm still seeing the instruments, is it ok?.

IMO the info of stopped should be moved to debug and if an exception happens (could not stop an element ) this should be reported instead (report only unexpected behaviors, IMO this improve the user experiences), since users know what they are stopping. 

An interesting feature will be to have the latest debug message (e.g. from last macro) for a post mortem study.   

```
Door_cfalcon_1 [1]: %ascan mot45 1 10 10 1
ScanDir is not defined. This operation will not be stored persistently. Use "expconf" or "newfile" to configure data storage (or eventually "senv ScanDir <abs directory>")
Scan #2 started at Wed Sep  1 07:31:57 2021. It will take at least 0:00:13.966124
 #Pt No    mot45      ct45      ct46      ct47      ct48       dt   
   0         1         1         2         3         4      0.298608
^C
Ctrl-C received: Stopping...
Stopping Motion(['mot45']) reserved by ascan
Motion(['mot45']) stopped
Stopping mntgrp08 reserved by ascan
mntgrp08 stopped
Stopping /monitor reserved by ascan
Stopping /slit reserved by ascan
Stopping /mirror reserved by ascan
   1        1.9        0      0.375574     0         0      1.61328 
Scan #2 ended at Wed Sep  1 07:31:59 2021, taking 0:00:01.834530. Dead time -9.0% (motion dead time 31.5%)
Executing ascan.on_stop method...
Stopping done!

Door_cfalcon_1 [3]: %lsi
       Name         Type   Parent
 ---------- ------------ --------
    /mirror   Instrument    -----
   /monitor   Instrument    -----
      /slit   Instrument    -----
```

**Comment by cmft** [comment]
Sorry @reszelaz,  I can not reopen it

**Comment by reszelaz** [comment]
I worked with @cmft on this issue.

First, his tests with the develop branch must still have used some outdated code. When fully using the develop branch there are only messages about stopping/aborting truly reserved objects (not instruments).

Second, the messages logged in two phases: _Stopping mntgrp08 reserved by ascan_ and  _mntgrp08 stopped_ were added on a special request #34. The motivation behind them is to give feedback to the user on what is going on during the stopping/aborting process which may take some time to complete, and on some occasions of not well-written controllers or faulty hardware may even get hung. Progressively seeing what is being stopped/aborted is IMO more useful than harmful. Not having this feedback may cause the impatient user to pass from the stopping phase to the aborting phase (2nd Ctrl+C) or even to the releasing phase (3rd Ctrl+C) what may have bad consequences. All this is documented in [Spock -> Stopping macros](https://sardana-controls.org/users/spock.html#stopping-macros).

When discussing it, @cmft proposed a nice improvement of not logging more scan records after macro was stopped/aborted. This way they won't mix with the stopping/aborting messages. I think this is a nice idea and I even tried to implement it however it is not so trivial. In the case of a step scan, we could easily determine if a macro was stopped/aborted before adding a record to the `RecordList`
```
if not self.macro.isStopped() and not self.macro.isAborted():
    self.data.addRecord(data_line)
```
but in a continuous scan, since we are processing data events in a worker thread we would need to:
- stop adding new jobs to the worker thread
- clear the queue of all not processed jobs
- set the `RecordList` class to not call `addRecord()` method when the worker thread executes the `addData()` method.

This works more or less well, but is not perfect anyway, since for both scans we are still pushing the output events in another worker thread. So, there is still a chance that the record messages will arrive to the client and mix with the stopping/aborting messages. 

In summary, even if the idea is nice, it is very hard to reliably implement within the current architecture.

So, I propose to present this issue to the users with the #1577 and #1579 fixed and explain the advantages of stopping/aborting messages. If after that there are still negative feedback, then please re-open again.

Many thanks!
